core/Settings/FieldConfig.php

```diff
@@ -214,6 +214,19 @@ class FieldConfig
      */
     public $transform = null;
 
+    /**
+     * A closure that prepares the setting value when loaded from storage.
+     *
+     * **Example**
+     *
+     *     $setting->prepareValue = function ($value, Setting $setting) {
+     *         return split(',', $value);
+     *     }
+     *
+     * @var null|\Closure
+     */
+    public $prepareValue = null;
```

Comment:
Not sure if it's better or worse (you can be the judge), but what about naming this variable `$decodeValue`?